code/modules/mob/living/simple_animal/hostile/hostile2.dm

```diff
@@ -169,6 +169,7 @@
 	LostTarget()
 
 /mob/living/simple_animal/hostile/proc/Goto(var/target, var/delay, var/minimum_distance)
+	step_towards(src, target) //weird but necessary so they try to bump openable obstacles
```

Comment:
It won't override retreat distance (so the distance that forces retreat), because Goto isn't called **at** retreat distance, so if it got called, there's at least one step from retreat distance:

```
			if(target_distance <= retreat_distance)//If target's closer than our retreat distance, run
				walk_away(src,target,retreat_distance,move_to_delay)
			**else
				Goto(target,move_to_delay,minimum_distance)**//Otherwise, get to our minimum distance so we chase them
```

They are already prone to 'cycling' without this commit (in testing, I discovered it can be used for an interesting hit-and-run melee behavior if you set retreat higher than minimal), but I threw in a min_dist check in there. They won't open doors if at minimum distance right now, but they won't cycle.

I would have done it with get_step_towards, but the fucking thing uses shitty built-in BYOND procs with no documentation on its architecture so I can't emulate its failure behavior well enough and I don't want to make snowflake checks for doors.